src/main/java/org/jenkinsci/plugins/authorizeproject/AuthorizeProjectProperty.java

```diff
@@ -119,15 +135,60 @@ public Authentication authenticate(Queue.Item item) {
         }
         return strategy.authenticate(owner, item);
     }
-    
-    @Initializer(after=InitMilestone.PLUGINS_STARTED)
+
+    /**
+     * Ensure that deserialization failures in this field result in a failure to deserialize the job.
+     * This method is responsible for ensuring that POSTing config.xml respects the defined strategy.
+     */
+    @Initializer(after = InitMilestone.PLUGINS_STARTED)
     public static void setStrategyCritical() {
         Items.XSTREAM2.addCriticalField(AuthorizeProjectProperty.class, "strategy");
     }
-    
+
+    /**
+     * If we are being deserialized outside of loading the initial jobs (or reloading) then we need to cross check
+     * the strategy permissions to defend against somebody trying to push a configuration relating to a user other
+     * than themselves.
+     *
+     * @return {@code this}
+     * @throws ObjectStreamException if the object cannot be deserialized.
+     * @see #setStrategyCritical()
+     */
+    private Object readResolve() throws ObjectStreamException {
+        if (strategy != null) {
+            Authentication authentication = Jenkins.getAuthentication();
+            if (authentication != ACL.SYSTEM) {
+                if (ProjectQueueItemAuthenticator.isConfigured()) {
+                    StaplerRequest request = Stapler.getCurrentRequest();
+                    AccessControlled context;
+                    if (request == null) {
+                        context = Jenkins.getActiveInstance();
+                    } else {
+                        Job job = request.findAncestorObject(Job.class);
+                        context = job == null ? Jenkins.getActiveInstance() : job;
+                    }
+                    try {
+                        strategy.checkConfigurePermission(context);
+                    } catch (AccessDeniedException e) {
+                        throw new InvalidObjectException(e.getMessage());
+                    }
+                }
+            }
+        }
+        return this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public JobProperty<?> reconfigure(StaplerRequest req, JSONObject form) throws Descriptor.FormException {
+        return strategy != null && ProjectQueueItemAuthenticator.isConfigured() ? this : null;
+    }

```

Comment:
(Note for myself)
* This is called when job configuration pages are submitted.
* authorize-project is no longer updated in job configuration pages.